### ISSUE TITLE
[Bug]: Missing headers are reported when using the SSO application

### ISSUE THREAD

**Issue by oculos** [nextcloud/server#44234]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Hi,

I'm running NC 29.0.0 beta 3.

I am getting this error: 

```
Some headers are not set correctly on your instance - The `X-Robots-Tag` HTTP header is not set to `noindex,nofollow`. This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly. - The `X-Permitted-Cross-Domain-Policies` HTTP header is not set to `none`. This is a potential security or privacy risk, as it is recommended to adjust this setting accordingly. 
```

The thing is that the headers are correctly set.

I am using the `user_saml` (SSO) application, and when I disable it, the error is gone. I realize that it's testing the root, because of the redirect to IdP, it's failing to see the headers that aren't set there.

Maybe the test should exclude redirects to the idp? 

### Steps to reproduce

1. Have Saml configured to an external idp
2. Missing headers error will show
3. Disable sso
4. No complaints about headers

### Expected behavior

It should only test for missing headers on its own domain, not on the idp.

### Installation method

Community Docker image

### Nextcloud Server version

master

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Nginx

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [X] SSO - SAML
- [ ] Other

**Comment by silmarine** [comment]
I have the same issue using traefik as my reverse proxy. I tried setting the settings in the nextcloud traefik labels but it didn't clear any of them for me. Following is my current traefik labels for nextcloud:
```
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.nextcloud.entrypoints=https"
      - "traefik.http.routers.nextcloud.rule=Host(`nextcloud.example.tld`)"
      - "traefik.http.routers.nextcloud.tls=true"
      - "traefik.http.routers.nextcloud.middlewares=nextcloudHeader"
      - "traefik.http.middlewares.nextcloudHeader.headers.stsSeconds=15552000"
      - "traefik.http.middlewares.nextcloudHeader.headers.forceSTSHeader=true"
      - "traefik.http.middlewares.nextcloudHeader.headers.frameDeny=false"
      - "traefik.http.middlewares.nextcloudHeader.headers.customFrameOptionsValue=sameorigin"
      - "traefik.http.middlewares.nextcloudHeader.headers.customResponseHeaders.X-Robots-Tag=noindex,nofollow"
     ```

**Comment by isdnfan** [comment]
I can confirm the problem - on NC29.0.1 the problem disappears once user_oidc is deactivated. 
I wold suggest checking some resource which doesn't require authentication e.g. `/status.php` to avoid redirection to IdP.

problem reported in help forum as well https://help.nextcloud.com/t/http-headers-warning/194499

**Comment by Ra72xx** [comment]
The Nextcloud security check site, which has no authentication to the respective Nextcloud installation at all, correctly detects the headers. So the check is possible without access.

**Comment by EwenBara** [comment]
I check the code and the check try to load the path "/heartbeat" (cf [here](https://github.com/nextcloud/server/blob/master/apps/settings/lib/SetupChecks/SecurityHeaders.php#L43C46-L43C55).

But this route seem does not exist. I guess the function resolve it to "/" as fallback. When I add the route as explain [here](https://help.nextcloud.com/t/unable-to-generate-a-url-for-the-named-route-heartbeat-nc-29/194816/6) headers check pass.

I don't know if it's the best solution but it's work.

**Comment by ramphex** [comment]
Same issue here with OIDC configured, no issue without OIDC.

**Comment by flotpg** [comment]
Same here with Azure AD / Entra ID SAML

**Comment by oculos** [comment]
Any progress on this? I just upgraded to 31.0.4, and I still see this error.

**Comment by isdnfan** [comment]
I can confirm the issue was fixed and my 2 instances (31.0.4) pass all setup checks
